app/controllers/admin/settings_controller.rb

```diff
@@ -17,24 +17,30 @@ def update
 
   # Add a new setting item
   def create
-    @setting = Setting.new( setting_params )
+    setting = Setting.new( setting_params )
 
-    if @setting.save
+    if setting.save
       flash[ :notice ] = I18n.t 'setting_created'
     else
       flash[ :alert ] = I18n.t 'setting_create_failed'
     end
     redirect_to admin_settings_path
+  rescue ActiveRecord::NotNullViolation, ActiveRecord::RecordNotFound
```

Comment:
What are the conditions that an `ActiveRecord::RecordNotFound` error happens in `create`? Similarly for the `NotNullViolation`. I'm wondering if this rescue is necessary.

Reply:
Not null violation is if you try to create one with no name, iirc. RecordNotFound doesn't make any sense for a create though, does it.

Reply:
Hrm, or maybe not. I must have got it mixed up with the update notnull - I shall remove it forthwith! :)